File: 0380-insert-delete-getrandom-o1/0380-insert-delete-getrandom-o1.java
class RandomizedSet {
    Set<Integer> set;
    ArrayList<Integer> array;
    public RandomizedSet() {
        set = new HashSet<>();
        array = new ArrayList<>();
    }
    
    public boolean insert(int val) {
        if(!set.contains(val)){
            set.add(val);
            array.add(val);
            return true;
        }
        return false;
    }
    
    public boolean remove(int val) {
        if(set.contains(val)){
            set.remove(val);
            int index = array.indexOf(val);
            array.remove(index);
            return true;
        }
        return false;
    }
    
    public int getRandom() {
        int max = array.size() - 1, min = 0;
        int randomNum = (int)(Math.random() * (max + 1));
        int element = array.get(randomNum);
        return element;
    }
}

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * RandomizedSet obj = new RandomizedSet();
 * boolean param_1 = obj.insert(val);
 * boolean param_2 = obj.remove(val);
 * int param_3 = obj.getRandom();
 */

File: 0380-insert-delete-getrandom-o1/0380-insert-delete-getrandom-o1.cpp
class RandomizedSet {
private:
    set<int> set;
    vector<int> vec;
public:
    RandomizedSet() {}
    bool insert(int val) {
        if(set.count(val)){
            return false;
        }
        set.insert(val);
        vec.push_back(val);
        return true;
    }
    
    bool remove(int val) {
        if(set.count(val)){
            set.erase(val);
            auto it = find(vec.begin(), vec.end(), val);
            int index = it - vec.begin();
            vec.erase(vec.begin() + index);
            return true;
        }
        return false;
    }
    
    int getRandom() {
        int max = vec.size() - 1;
        int randomNum = rand() % (max + 1);
        return vec[randomNum];
    }
};

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * RandomizedSet obj = new RandomizedSet();
 * boolean param_1 = obj.insert(val);
 * boolean param_2 = obj.remove(val);
 * int param_3 = obj.getRandom();
 */